### DISCUSSION TITLE
Do you have any advice on how to handle existing plugin blocks when a feature that was implemented in a plugin is implemented in the core?

### DISCUSSION THREAD

**Discussion by kohheepeace** [WordPress/gutenberg#58290] [General]
It's just like the title says,

**_Do you have any advice on how to handle existing plugin blocks when a feature that was implemented in a plugin is implemented in the core?_**

For example, the core currently has the following new block PRs/issues

- https://github.com/WordPress/gutenberg/pull/47765
- https://github.com/WordPress/gutenberg/issues/21584

But consider if this functionality was already provided by a plugin.

Do you have any advice on migration, etc... in this case? Sorry for the sketchy question. 😢

@Mamaduka any thoughts about your plugin ?
- https://github.com/Mamaduka/bookmark-card

**Comment by Mamaduka** [comment]
You can continue supporting your existing plug-in if it has different enough features. I might consider doing that for a Bookmark Card.

Other options:
* Provide a [transformation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-transforms/) option to the new core block. This requires explicit action from the user, and they can remove the plug-in once all transformations are done.
* Use the [`useMigrateOnLoad`](https://github.com/WordPress/gutenberg/blob/f5917593d17f2f7f5cedee4cd2e0e89b88b71f3a/packages/block-library/src/list/edit.js#L43) technique. It will migrate the existing block to a new one, a post containing that block. This is not an explicit action, and not all users might want to migrate, so be considerate when implementing it.

**Comment by kohheepeace** [comment]
Super thanks for your concise and clear answer! It is very helpful 💯
